server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamFailureStore.java

```diff
@@ -29,6 +29,7 @@
  * supports the following configurations only explicitly enabling or disabling the failure store
  */
 public record DataStreamFailureStore(Boolean enabled) implements SimpleDiffable<DataStreamFailureStore>, ToXContentObject {
+    public static final String FAILURES_LIFECYCLE_API_CAPABILITY = "failures_lifecycle";
```

Comment:
Nit: It looks like this is only used in the rest action. Does it make sense to move this there? Alternatively, it's describing something about the structure of the response, should it be moved to the response object/action class?

Reply:
Also, name wise, should we make this something like `failure_store.lifecycle` to denote specifically what part of the request is new? I see we have other changes to the response as well, so this is a soft suggestion. If you think this should be more generic to describe those changes that's also fine.

Reply:
+1 on the naming, about placement, I put it here because it will be used also by the data stream options later. But we can always move it then.